src/hooks/useFormCollapse.tsx

```diff
@@ -0,0 +1,43 @@
+import { SessionMode } from '../types';
+import { useEffect, useState } from 'react';
+
+export function useFormCollapse(sessionMode: SessionMode) {
```

Comment:
This looks over engineered. I would opt for passing this state to the `FormEngine` component within the `esm-form-engine-app` and use it to toggle the sections